Data/ByteString.hs

```diff
@@ -697,15 +707,18 @@ scanl :: (Word8 -> Word8 -> Word8) -> Word8 -> ByteString -> ByteString
 scanl f v (BS fp len) = unsafeDupablePerformIO $ withForeignPtr fp $ \a ->
     create (len+1) $ \q -> do
         poke q v
-        scanl_ v 0 a (q `plusPtr` 1)
+        go a q
```

Comment:
I would just pass `q `plusPtr` 1` as the second argument to `go` here, and after renaming `(p, ptr)` to `(src, dst)` as above, just use those in `go`.